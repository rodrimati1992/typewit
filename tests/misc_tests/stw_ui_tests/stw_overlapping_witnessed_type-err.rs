
typewit::simple_type_witness!{
    enum TwoU8s {First = u8, Second = u8 }
}

typewit::simple_type_witness!{
    enum OrConcrete<T> {Vectu8 = Vec<u8>, Generic = T }
}



fn main() {}