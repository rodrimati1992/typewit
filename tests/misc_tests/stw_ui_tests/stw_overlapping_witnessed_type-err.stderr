error[E0119]: conflicting implementations of trait `MakeTypeWitness` for type `TwoU8s<u8>`
 --> tests/misc_tests/type_eq_ui_tests/stw_overlapping_witnessed_type-err.rs:2:1
  |
2 | / typewit::simple_type_witness!{
3 | |     enum TwoU8s {First = u8, Second = u8 }
4 | | }
  | | ^
  | | |
  | |_first implementation here
  |   conflicting implementation for `TwoU8s<u8>`
  |
  = note: this error originates in the macro `$crate::__impl_with_span` which comes from the expansion of the macro `typewit::simple_type_witness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `MakeTypeWitness` for type `OrConcrete<Vec<u8>, Vec<u8>>`
 --> tests/misc_tests/type_eq_ui_tests/stw_overlapping_witnessed_type-err.rs:6:1
  |
6 | / typewit::simple_type_witness!{
7 | |     enum OrConcrete<T> {Vectu8 = Vec<u8>, Generic = T }
8 | | }
  | | ^
  | | |
  | |_first implementation here
  |   conflicting implementation for `OrConcrete<Vec<u8>, Vec<u8>>`
  |
  = note: this error originates in the macro `$crate::__impl_with_span` which comes from the expansion of the macro `typewit::simple_type_witness` (in Nightly builds, run with -Z macro-backtrace for more info)
